app/models/submission/file.rb

```diff
@@ -13,7 +13,7 @@ class Submission::File < ApplicationRecord
   # any content that has been passed in via the
   # attr_writer. We only want to do this on create
   # and never want to update or change this content.
-  after_create do
+  after_create_commit do
```

Comment:
I think we don't need it here, as the code inside seems to only use the current model's values. I still changed it to `after_create_commit` to prevent any possible future mistakes. Let me know what you think.

Reply:
Updated